catalyst-toolbox/src/rewards/veterans.rs

```diff
@@ -77,13 +77,35 @@ fn calc_final_eligible_rankings(
         .collect()
 }
 
+fn calc_final_ranking_consensus_per_review(rankings: &[impl Borrow<VeteranRankingRow>]) -> Decimal {
+    let rankings_majority = Decimal::from(rankings.len()) / Decimal::from(2);
+    let ranks = rankings.iter().counts_by(|r| r.borrow().score());
+
+    match (ranks.get(&FilteredOut), ranks.get(&Excellent), ranks.get(&Good)) {
+        (Some(filtered_out), _, _) if Decimal::from(*filtered_out) >= rankings_majority => {
+            Decimal::from(*filtered_out) / Decimal::from(rankings.len())
+        }
+        (_, Some(excellent), _) if Decimal::from(*excellent) > rankings_majority => {
+            Decimal::from(*excellent) / Decimal::from(rankings.len())
+        }
+        (_, Some(excellent), Some(good)) => {
+            (Decimal::from(*excellent) + Decimal::from(*good)) / Decimal::from(rankings.len())
+        }
+        (_, _, Some(good)) => {
```

Comment:
nit: this can be grouped with the arm above by doing
```
(_, maybe_excellent, Some(good)) => Decimal::from(maybe_excellent.unwrap_or_default()) + ...
```

Reply:
I had to do a map to transform our `maybe_excellent`  (Option[&usize]) into an  Option[Decimal] before I could use the unwrap_or_default() on it:

```Rust
(_, maybe_excellent, Some(good)) => {
            (maybe_excellent.map(|excellent| Decimal::from(*excellent)).unwrap_or_default() + Decimal::from(*good)) / Decimal::from(rankings.len())
        }
```

As I said I'm not familiar with Rust at all so there might be a better way... (I'm very used to Scala)

Reply:
fixed by 74a5544, tell me if there is a better way

Reply:
ah I see, `Option` of a reference type has no default, but this should work
```
(_, maybe_excellent, Some(good)) => Decimal::from(maybe_excellent.copied().unwrap_or_default()) + ...

```
`copied()` turns `Option<&usize>` into `Option<usize>` by copying the value

Reply:
thanks, it's fixed in 3917c88